src/gui/qgstemporalcontrollerwidget.cpp

```diff
@@ -574,7 +575,12 @@ void QgsTemporalControllerWidget::setDates( const QgsDateTimeRange &range )
     whileBlocking( mEndDateTime )->setDateTime( range.end() );
     whileBlocking( mFixedRangeStartDateTime )->setDateTime( range.begin() );
     whileBlocking( mFixedRangeEndDateTime )->setDateTime( range.end() );
-    updateTemporalExtent();
+    // only if range is different from mNavigationObject->temporalExtents update
+    // else recursion as updateTemporalExtent itself calls mNavigationObject->setTemporalExtents
+    if ( range != mNavigationObject->temporalExtents() )
```

Comment:
@rduivenvoorde is this check still required / meaningful after the recursion guard here: https://github.com/qgis/QGIS/pull/38601/files#diff-dda9bd3113bde0d8f7a3c8f5e3039842R131-R134